lib/fs.js

```diff
@@ -1444,12 +1477,21 @@ function readdir(path, options, callback) {
  * @param {string | {
  *   encoding?: string;
  *   withFileTypes?: boolean;
+ *   recursive?: boolean;
  *   }} [options]
  * @returns {string | Buffer[] | Dirent[]}
  */
 function readdirSync(path, options) {
   options = getOptions(options);
   path = getValidatedPath(path);
+  if (options.recursive != null) {
+    validateBoolean(options.recursive, 'options.recursive');
```

Comment:
Do we have to be strict about `typeof options.recursive === 'boolean'`, or we can allow any truthy/falsy value, similar to `options.withFileTypes`?

Reply:
> Not being strict with boolean type checks create issues where you also have to check `1`.

I'm not sure if I follow. Loose typecheck on positional arguments can be error-prone in some scenarios, but I don't see an issue in this form. Can you provide an example?
If the logic will be moved to C++ layer, we can easily pass `!!recursive` to it, similar to how it's done for `withFileTypes` in the same function:
https://github.com/nodejs/node/blob/611a616a8a27458775e8c235fe4faea2b628963c/lib/fs.js#L1496-L1498
If strict check is preferable, perhaps we shouldn't allow `null` value either, and align other boolean parameters with it.

Reply:
I see. I recommend @Ethan-Arrowood to follow up with this on a different pull request, if it's ok with you?

Reply:
This is a part of stable API, so changing it separately might be `semver-major`.

That said, I'm not blocking this PR in current form, but it would be nice to have consensus on what we want to have after follow-ups before landing this one.
Since recursive `readdir` can take a long time, I think ideally we can move validation/coercion to `getOptions()` and make it return a copy rather than original object. This will make it unaffected by mutations of `options` while it's still running, and align other methods with same behavior.